### ISSUE TITLE
Missing DigiCert SHA2 Extended Validation Server CA in current cacert.pem download

### ISSUE THREAD

**Issue by wmjrick** [bagder/curl#570]
Vendor that our company interfaces with recently renewed SSL cert with DigiCert.
Subsequent connections to their site using curl resulted in validation error.
Systems in question are running curl on RHEL 5.8.
Downloaded latest bundle using curl:    curl http://curl.haxx.se/ca/cacert.pem  -o /etc/pki/tls/certs/ca-bundle.crt
Retested and validation error still occurred.
Reviewed validation path and found that DigiCert SHA2 Extended Validation Server CA was missing.
Exported from Windows version of Firefox and appended to this latest cacert.pem.
Connections to vendor site now functioning.

**Comment by jay** [comment]
TL;DR: The problem could be your vendor's server not sending that intermediate certificate.

I reviewed [the raw certificate lists](https://github.com/bagder/curl/blob/master/lib/mk-ca-bundle.pl#L47-L58) that the bundle is made from but I do not see 'DigiCert SHA2 Extended Validation Server CA' included by any of them. This is likely because it is an intermediate certificate, not a root. Its root is 'DigiCert High Assurance EV Root CA', which is in the raw info (and therefore is in the ca-bundle).

Typically a server will supply intermediates. My guess is your vendor's server does not supply that intermediate. In Firefox if it receives a valid intermediate it will cache the certificate. You can tell the difference between the two storage types because one will show as 'Builtin Object Token' (ie default certs eg [ca-bundle.crt](https://github.com/bagder/ca-bundle)) and the other as 'Software Security Device' (ie the certs Firefox cached).

So imagine some other server (take https://www.digicert.com for example -- `openssl s_client -showcerts -connect www.digicert.com:443`) has already supplied the Digicert SHA2 intermediate which is then cached by Firefox, and when you later try to access your vendor's server in Firefox --the server which does not supply the intermediate-- there is no problem because it already has it in its cache. *You're just lucky it happens that way.*

libcurl does not cache intermediate certificates but the SSL backend it's using may. For WinSSL the intermediates are cached. For OpenSSL they are not. NSS, DarwinSSL, etc, I don't know.

If this is the case contact your vendor because their server should send the intermediate. If this is not the case we will need more information (curl -V) and if the vendor's server is public its info so we can try to reproduce.

**Comment by jay** [comment]
Closing due to lack of response

**Comment by tvt1905** [comment]
Hi
Vào 02-01-2016 09:47, "Jay Satiro" <notifications@github.com> đã viết:

**Comment by patrikhson** [comment]
Hey,
We've got some great news for you! Just take a look <http://stebeluke.theoakhurstdoctor.com/xyojwg>

Kind regards, paf@frobbit.se
